src/main/kotlin/com/autonomousapps/tasks/ProjectHealthTask.kt

```diff
@@ -15,6 +15,8 @@ import org.gradle.api.tasks.TaskAction
 
 abstract class ProjectHealthTask : DefaultTask() {
 
+  private val buildFilePath = project.buildFile.path
```

Comment:
The `project` (`getProject()`) cannot be executed inside of `@TaskAction` for tasks that are not explicitly marked as non-cacheable:
```
1 problem was found storing the configuration cache.
- Task `:proj:projectHealth` of type `com.autonomousapps.tasks.ProjectHealthTask`: invocation of 'Task.project' at execution time is unsupported.
  See https://docs.gradle.org/7.4/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution
```
So this simple trick is used to resolve the project parameter on the phase before TaskAction execution and not to pass as Input from the plugin. You suggested to make it an input - okay. Initially I didn't mark it as input, because this task is not cacheable.